### DISCUSSION TITLE
inline-source-map generating separate map file

### DISCUSSION THREAD

**Discussion by rexkenley** [nrwl/nx#28416] [Questions]
I have this webpack config file that is generating a separate map file. The map should be bundled with the main file when using inline-source-map devtool, correct?

common
![image](https://github.com/user-attachments/assets/335569ee-201b-4c9f-b58c-5067c8270c36)

dev
![image](https://github.com/user-attachments/assets/ff675a7b-fb69-4e4e-aa8f-23f5b91657b8)

result
![image](https://github.com/user-attachments/assets/d5c9f27f-1baf-4c41-af99-e02ab8cb8c27)

**Comment by ndcunningham** [comment]
Hey, thanks for the question.

By default, if `sourceMaps` is not defined via `NxAppWebpackPlugin` sourcemaps should be set to false. This means, there will be no sourceMaps generated. https://github.com/nrwl/nx/blob/42da5421afdd21700486fbb1e26d22e8f36bc199/packages/webpack/src/plugins/nx-webpack-plugin/lib/apply-base-config.ts#L88C1-L93C15

One way you could handle this is by creating your plugin or stub the functionality.  
This would be similar to if you want to override `NODE_ENV` set by `DefinePlugin`.

```
  plugins: [
    new NxAppWebpackPlugin({
      tsConfig: './tsconfig.app.json',
      compiler: 'babel',
      main: './src/main.tsx',
      index: './src/index.html',
      baseHref: '/',
      assets: ['./src/favicon.ico', './src/assets'],
      styles: ['./src/styles.css'],
      outputHashing: process.env['NODE_ENV'] === 'production' ? 'all' : 'none',
      optimization: process.env['NODE_ENV'] === 'production',
      // sourceMap: 'source-map',
    }),
    new NxReactWebpackPlugin({
      // Uncomment this line if you don't want to use SVGR
      // See: https://react-svgr.com/
      // svgr: false
    }),
    //  <--- Add your plugin here
  ]
```
  We should extend the `sourceMap` option so that you can also pass the value you want.
